version: "3.8"

services:
  mongo1:
    image: mongo:7.0.11
    command: ["--replSet", "rs0", "--bind_ip_all", "--port", "27017"]
    ports:
      - "27017:27017"
    extra_hosts:
      - "host.docker.internal:host-gateway"
    healthcheck:
      test: echo "try { rs.status() } catch (err) { rs.initiate({_id:'rs0',members:[{_id:0,host:'host.docker.internal:27017',priority:1},{_id:1,host:'host.docker.internal:27018',priority:0.5}]}) }" | mongosh --port 27017 --quiet
      interval: 5s
      timeout: 30s
      start_period: 0s
      # noinspection ComposeUnknownKeys
      start_interval: 1s
      retries: 10
    volumes:
      - "mongo1_data:/data/db"
      - "mongo1_config:/data/configdb"
    deploy:
      resources:
        limits:
          memory: 512M

  mongo2:
    image: mongo:7.0.11
    command: ["--replSet", "rs0", "--bind_ip_all", "--port", "27018"]
    ports:
      - "27018:27018"
    extra_hosts:
      - "host.docker.internal:host-gateway"
    volumes:
      - "mongo2_data:/data/db"
      - "mongo2_config:/data/configdb"
    deploy:
      resources:
        limits:
          memory: 512M


volumes:
  mongo1_data:
  mongo2_data:
  mongo1_config:
  mongo2_config:
